### PR TITLE
Provide ONNX external data mechanism to ReadNetwork

### PR DESCRIPTION
**Pull request by mbencer**
It is continuation of: https://github.com/openvinotoolkit/openvino/pull/2303

### CODE REVIEW

**Review by tsocha — APPROVED** [review]

File: inference-engine/src/inference_engine/ie_network_reader.hpp [hunk 1]
```diff
@@ -26,6 +26,7 @@ CNNNetwork ReadNetwork(const std::string& modelPath, const std::string& binPath,
  * @param model string with IR
  * @param weights shared pointer to constant blob with weights
  * @param exts vector with extensions
+ * @note If you read onnx model which requires external data feature, use ReadNetwork overloading which takes path to the model.
```
Comment (postrational) [inline inference-engine/src/inference_engine/ie_network_reader.hpp:29]:
```suggestion
 * @note Reading ONNX models doesn't support loading weights from data blobs.
         If you are using an ONNX model with external data files, please use the
         ReadNetwork function overload which takes a filesystem path to the model.
```

File: ngraph/frontend/onnx_import/include/onnx_import/utils/tensor_external_data.hpp [hunk 2]
```diff
@@ -33,7 +33,7 @@ namespace ngraph
                 /// \brief      Load external data from tensor passed to constructor
                 ///
                 /// \note       If read data from external file fails,
-                ///             the invalid_external_data is thrown
+                ///             the invalid_external_data is thrown.
```
Comment (postrational) [inline ngraph/frontend/onnx_import/include/onnx_import/utils/tensor_external_data.hpp:36]:
```suggestion
                /// \note       If reading data from external files fails,
                ///             the invalid_external_data exception is thrown.
```

**Review by postrational — APPROVED** [review]
Looks good, but I have a few suggestions to comments.

File: inference-engine/include/ie_core.hpp [hunk 3]
```diff
@@ -79,6 +77,8 @@ class INFERENCE_ENGINE_API_CLASS(Core) {
      * @param model string with model in IR or ONNX format
      * @param weights shared pointer to constant blob with weights
      * ONNX models doesn't support models with data blobs.
+     * If you read ONNX model which requires external data feature,
+     * use ReadNetwork overloading which takes path to the model.
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_core.hpp]: can we refer to exact overloading? E.g. `use InferenceEngine::Core::ReadNetwork(const std::string& model, const Blob::CPtr& weights) const`

File: inference-engine/include/ie_core.hpp [hunk 4]
```diff
@@ -70,15 +71,18 @@ class INFERENCE_ENGINE_API_CLASS(Core) {
...
      */
     CNNNetwork ReadNetwork(const std::string& modelPath, const std::string& binPath = {}) const;
     /**
      * @brief Reads models from IR and ONNX formats
      * @param model string with model in IR or ONNX format
      * @param weights shared pointer to constant blob with weights
-     * ONNX models doesn't support models with data blobs.
+     * Reading ONNX models doesn't support loading weights from data blobs.
+     * If you are using an ONNX model with external data files, please use the
+     * ReadNetwork function overload which takes a filesystem path to the model.
```
Comment (ilya-lavrenov) [inline inference-engine/include/ie_core.hpp:85]: can we refer to exact overloading? E.g. `use InferenceEngine::Core::ReadNetwork(const std::string& model, const Blob::CPtr& weights) const`. you can build docs via make openvino_docs and see how it looks like

Comment (mbencer) [reply inference-engine/include/ie_core.hpp]: @ilya-lavrenov done, the proper link was generated.
image

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]
LGTM. Am I right that previously ONNX importer didn't support unicode before this PR?

**Comment by mbencer** [comment]
> LGTM. Am I right that previously ONNX importer didn't support unicode before this PR?
@ilyachur 
That's right, but in current version OnnxReader uses only `import_onnx_model` function overloading which takes stream so for importing model should work for unicode path too.
In order to provide unicode path support for external data feature I have to extend some utils of nGraph.
However, only model part of path to external file can be unicode, because relative paths saved in onnx model can be only single-byte (it is limitation of protobuf).

File: inference-engine/tests/functional/inference_engine/CMakeLists.txt [hunk 5]
```diff
@@ -33,6 +33,7 @@ ie_faster_build(${TARGET_NAME}
 )
 
 if(TARGET inference_engine_onnx_reader)
+    add_definitions("-DONNX_TEST_MODELS=\"${CMAKE_CURRENT_SOURCE_DIR}/onnx_reader/models/\"")
```
Comment (ilya-lavrenov) [inline inference-engine/tests/functional/inference_engine/CMakeLists.txt:36]: `add_definitions` -> `target_compile_definitions`

**Review by tomdol — APPROVED** [review]

File: inference-engine/tests/functional/inference_engine/CMakeLists.txt (hunk 5 above)
Comment (mbencer) [reply inference-engine/tests/functional/inference_engine/CMakeLists.txt]: done

**Review by mikhail-treskin — APPROVED** [review]

**Comment by ilyachur** [comment]
Please rebase on the last master

**Comment by ilyachur** [comment]
Please add information about ONNX reader limitations to next guides:

- https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_DG/protecting_model_guide.md

- https://github.com/openvinotoolkit/openvino/blob/master/docs/IE_DG/ONNX_Support.md

I think we can do it separately.

**Pull request merged**